src/fsharp/FSharp.Core/array.fs

```diff
@@ -1136,23 +1128,22 @@ namespace Microsoft.FSharp.Collections
             [<CompiledName("Partition")>]
             let partition predicate (array : 'T[]) =
                 checkNonNull "array" array
-                let inputLength = array.Length
-                let lastInputIndex = inputLength - 1
+                let inputLength = array.Length                
 
                 let isTrue = Array.zeroCreateUnchecked inputLength
                 Parallel.For(0, inputLength, 
                     fun i -> isTrue.[i] <- predicate array.[i]
                     ) |> ignore
                 
                 let mutable trueLength = 0
-                for i in 0 .. lastInputIndex do
+                for i in 0 .. isTrue.Length-1 do
```

Comment:
I'm referring to the use of `for i in 0 .. bla` instead of `for i = 0 to bla`. There's no perf difference but you've used the latter consistently throughout the rest of the PR, so just thought it would make more sense.